File: bms/go/geo/geo_s1_Interval_InteriorContainsInterval/geo_s1_Interval_InteriorContainsInterval.go
package test

import (
	"math"
)

func (i geo_s1_Interval) InteriorContainsInterval(oi geo_s1_Interval) bool {
	if i.IsInverted() {
		if oi.IsInverted() {
			return (oi.Lo > i.Lo && oi.Hi < i.Hi) || oi.IsEmpty()
		}
		return oi.Lo > i.Lo || oi.Hi < i.Hi
	}
	if oi.IsInverted() {
		return i.IsFull() || oi.IsEmpty()
	}
	return (oi.Lo > i.Lo && oi.Hi < i.Hi) || i.IsFull()
}

func (i geo_s1_Interval) IsInverted() bool { return i.Lo > i.Hi }

func (i geo_s1_Interval) IsEmpty() bool { return i.Lo == math.Pi && i.Hi == -math.Pi }

func (i geo_s1_Interval) IsFull() bool { return i.Lo == -math.Pi && i.Hi == math.Pi }

type geo_s1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_s1_Interval_InteriorContainsInterval/mistral/Restart/geo_s1_Interval_InteriorContainsInterval.rs

use std::f64;

struct GeoS1Interval {
    lo: f64,
    hi: f64,
}

impl GeoS1Interval {
    fn is_inverted(&self) -> bool {
        self.lo > self.hi
    }

    fn is_empty(&self) -> bool {
        (self.lo == f64::consts::PI) && (self.hi == -f64::consts::PI)
    }

    fn is_full(&self) -> bool {
        (self.lo == -f64::consts::PI) && (self.hi == f64::consts::PI)
    }
}

fn interior_contains_interval(i: &GeoS1Interval, oi: &GeoS1Interval) -> bool {
    if i.is_inverted() {
        if oi.is_inverted() {
            (oi.lo > i.lo && oi.hi < i.hi) || oi.is_empty()
        } else {
            oi.lo > i.lo || oi.hi < i.hi
        }
    } else if oi.is_inverted() {
        i.is_full() || oi.is_empty()
    } else {
        (oi.lo > i.lo && oi.hi < i.hi) || i.is_full()
    }
}
